mscore/plugin/api/elements.cpp

```diff
@@ -69,6 +70,16 @@ void Note::setTpc(int val)
             set(Pid::TPC2, val);
       }
 
+//---------------------------------------------------------
+//   Chord::setPlayEventType
+//---------------------------------------------------------
+
+void Chord::setPlayEventType(Ms::PlayEventType v)
+      {
+      chord()->score()->setPlaylistDirty();
+      chord()->score()->undo(new ChangeChordPlayEventType(chord(), v));
```

Comment:
This function could potentially check whether the chord already has a correct play event type, then it could avoid pushing extra unnecessary commands to undo stack. That is not strictly necessary but could optimize this operation a bit.

Reply:
But isn't it possible that, even though the playEventType hasn't changed, the playEventLists may have? Woudn't those changes would be lost to a missing redo operation?

Reply:
Sorry, maybe I didn't understand the question properly. This function has nothing to do with changing note events at all, if a plugin needs to change play events it would modify the `Note.playEvents` property which would also change the play event type to `User`.

Reply:
Ok. I understand. Check is added.

-Dale